marshmallow/fields.py

```diff
@@ -1127,28 +1127,55 @@ def _deserialize(self, value):
 
 
 class TimeDelta(Field):
-    """Formats time delta objects, returning the total number of seconds
-    as a float.
+    """A field that (de)serializes a :class:`datetime.timedelta` object to an
+    integer and vice versa. The integer can represent the number of days,
+    seconds or microseconds.
 
+    :param str precision: Influences how the integer is interpreted during
+        (de)serialization. Must be 'days', 'seconds' or 'microseconds'.
+    :param str error: Error message stored upon validation failure.
     :param kwargs: The same keyword arguments that :class:`Field` receives.
     """
 
+    DAYS = 'days'
+    SECONDS = 'seconds'
+    MICROSECONDS = 'microseconds'
+
+    def __init__(self, precision='seconds', error=None, **kwargs):
+        precision = precision.lower()
+        units = (self.DAYS, self.SECONDS, self.MICROSECONDS)
+
+        if precision not in units:
+            msg = 'The precision must be "{0}", "{1}" or "{2}".'.format(*units)
+            raise ValueError(msg)
+
+        self.precision = precision
+        super(TimeDelta, self).__init__(error=error, **kwargs)
+
     def _serialize(self, value, attr, obj):
         try:
-            return total_seconds(value)
+            days = value.days
+            seconds = days * 86400 + value.seconds
+            microseconds = seconds * 10**6 + value.microseconds
```

Comment:
I wonder if we can save unneeded computation here by only computing to the requested precision. Something like:

```python
    def _serialize(self, value, attr, obj):
        try:
            days = value.days
            if self.precision == self.DAYS:
                return days
            else:
                seconds = days * 86400 + value.seconds
                if self.precision == self.SECONDS:
                    return seconds
                else:  # microseconds
                    return seconds * 10**6 + value.microseconds
        except AttributeError:
            msg = '{0!r} cannot be formatted as a timedelta.'.format()
            raise MarshallingError(getattr(self, 'error', None) or msg)
```

Perhaps there is a more elegant way, but it would be good remove the extra computation.